server/static/upload.js

```diff
@@ -1,137 +1,242 @@
+        $('.edit-sample-name').css('background-color', '#eee')
+        
+    });
+    
+    // Editing the sample name will remove the drop area
+    $('.edit-sample-name').on('click', function(e) {
+        $('#sample-text').prop('disabled', false);
+        $('.resumable-drop').hide();
+        $('.edit-sample-name').css('background-color', '#eee')
     });
-    $('.resumable-drop').on('drop', function() {
-        $(this).removeClass('resumable-dragover');
+    
+    // Remove file from add sample table
+    $('.table-data').on('click', '.remove-file', function(e) {
+        fileName = $(this).parents('tr').find('.file-name').text();
+        removeResumableFile(fileName);
+        $(this).parents('tr').remove();
+    });
+    
+    // Remove entire sample from add sample table
+    $('.table-data').on('click', '.remove-sample', function(e) {
+        var fileNames = []
+        
+        fileRows = $(this).parents('tr').nextUntil('.sample-header-template')
+        fileRows.each(function(key, value) {
+            fileNames.push($(value).find('.file-name').text())
+        });
+        for (i = 0; i < fileNames.length; i++) {
+            removeResumableFile(fileNames[i])
+        }
+        ;
+        fileRows.remove();
+        $(this).parents('tr').remove();
     });
 
-    $('body').on('click', '.progress-cancel-link', function() {
-        // Cancel upload of this file
-        var progress = $(this).closest('.resumable-progress');
-        var uniqueIdentifier = progress.attr('data-file-id');
-        r.getFromUniqueIdentifier(uniqueIdentifier).cancel();
-        // Remove row
-        progress.remove();
-        return false;
+    // Add files to a specific sample
+    $('.table-data').on('click', '.add-files', function(e) {
+        sampleName = $(this).parents('tr').find('.sample-name').text();
     });
-    $('body').on('click', '.progress-retry-link', function() {
-        // Retry upload of this file
-        var progress = $(this).closest('.resumable-progress');
-        var uniqueIdentifier = progress.attr('data-file-id');
-        r.getFromUniqueIdentifier(uniqueIdentifier).retry();
 
-        progress.find('.progress-bar').first()
-            .removeClass('progress-bar-warning progress-bar-danger')
-            .addClass('progress-bar-striped active')
-            .html('Retrying...');
-        progress.find('.progress-cancel-link').show();
-        progress.find('.progress-retry-link').hide();
-        return false;
+    // Edit a sample name in table
+    $('.table-data').on('click', '.edit-table-sample-name', function(e){
+        sampleNode = $(this).parents('tr').find('.sample-name');
+        currentName = sampleNode.text()
+        sampleNode.replaceWith(function(){
+            return $('<td class="sample-name"><form class=edit-sample-name-form>'+
+            '<input id="sample-table-text" type="text" name="sampleName" class="form-control"'+
+            'placeholder="'+currentName+'"></form></td>')
```

Comment:
This should have the value be the previous value, rather than setting the placeholder. You should probably set it via the `val()` method as well.